pkg/subctl/cmd/gather.go

```diff
@@ -192,11 +186,32 @@ func gatherDiscovery(dataType string, info *gather.Info) bool {
 func gatherBroker(dataType string, info *gather.Info) bool {
 	switch dataType {
 	case Logs:
-		info.Status.QueueWarningMessage("No logs to gather on Broker")
+		info.Status.QueueSuccessMessage("No pods accessible. No logs to collect on Broker")
 	case Resources:
-		_, _ = getBrokerRestConfig(info.Submariner)
+		brokerRestConfig, err := getBrokerRestConfig(info.Submariner)
+		if err != nil {
+			info.Status.QueueFailureMessage(fmt.Sprintf("Error getting Broker rest config %s", err))
+			return true
+		}
+
+		info.DynClient, info.ClientSet, err = getClients(brokerRestConfig)
```

Comment:
`info` is passed by reference so you can't modify unless you restore its previous state on return. But, as I mentioned earlier, you can simply pass by value instead, ie

`func gatherBroker(dataType string, info gather.Info) bool {`

Reply:
Not addressed this comment. Will do in coming update

Reply:
I guess this states otherwise https://goinbigdata.com/golang-pass-by-pointer-vs-pass-by-value/

Reply:
Passing a struct by pointer is obviously more efficient but it does allow the recipient to modify the struct and its effects are observed on return.  This may not be desirable and is not in this case. You are modifying fields of `info` which is subsequently passed to the next gather function and thus will see `ClusterName` as "broker" and `DynClient` pointing to the broker cluster.  You can see this effect in the E2E tests. Noticed they all failed, eg https://github.com/submariner-io/submariner-operator/pull/1187/checks?check_run_id=2239780706. After **Gathering broker resources**, it goes onto **Gathering operator resources** but that fails: "_Failed to gather submariners: error listing "submariners": submariners.submariner.io is forbidden: User "system:serviceaccount:submariner-k8s-broker:cluster-cluster2" cannot list resource "submariners" in API group "submariner.io" in the namespace "submariner-operator"_".  That is b/c `DynClient` pointing to the broker cluster which doesn't have the CRD.

So you must make a copy of `info`, hence my suggestion. Passing by value in this case is negligible wrt performance and automatically allows each gather function to safely modify the struct if need be w/o side-effects.

Reply:
yes. Its the `after gather broker` calls that are affected. I realised this much later after I had logged off for the day. I initially thought the concern is about the config changes `before gather broker` is called